### PR TITLE
fix issue with test failures in auto tests

### PR DESCRIPTION
**Pull request by mer-a-o**

### CODE REVIEW

**Comment by mer-a-o** [comment]
@mmiesch, @srherbener:
I updated intel19 container and it solved the issue with finding odc. After installing xarray in CodeBuild and Travis-CI all tests pass except `test_iodaconv_odb2nc_tool` in CodeBuild GNU container (not in Travis-CI GNU container). The paths look correct:
`PATH = /usr/local/bin:/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/codebuild/user/bin`
`PYTHONPATH  = /usr/local/lib`
`LD_LIBRARY_PATH = /usr/local/lib:`

But the test fails with this error: 
```
OSError: cannot load library 'libodccore.so': libodccore.so: cannot open shared object file: No such file or directory.  Additionally, ctypes.util.find_library() did not manage to locate a library called 'libodccore.so'
```
Any idea on how to solve this issue?

**Comment by srherbener** [comment]
@mer-a-o, not sure what's going on. I tried this on my Mac using the charliecloud intel19 container (dated Jun 8) and it works. The file /usr/local/lib/libodccore.so exists, which is where it should be. Looks like pyodc 1.0.2 is installed, which is correct.

The charliecloud container has python3.6.  I tried importing ctypes and there is not a util subpackage. Perhaps the python version is not right, or the ctypes package is not right. My ctypes version is 1.1.0.

**Comment by srherbener** [comment]
Sorry, tested the wrong container. Retried this using the Singularity gnu-openmpi container dated June 8. Same results. All tests pass, libodccore.so is in /usr/local/lib, pyodc is 1.0.2, python 3.6.

**Comment by mer-a-o** [comment]
@srherbener, yes everything passes inside the container but in CodeBuild the paths are somehow reset and I need to specify them again. We don't have this problem in Travis-CI so everything passes there. Is there any other path that I need to set other than `PATH`, `LD_LIBRARY_PATH`, and `PYTHONPATH` ?

**Comment by mer-a-o** [comment]
The issue is finally solved! I had to set a user (jcsdauser) to be able to run mpi jobs in the gnu container. In CodeBuild, I had to set the paths for jcsduser in the same command line as ctest.

**Comment by mmiesch** [comment]
Great work @mer-a-o!

File: .travis.yml [hunk 1]
```diff
@@ -56,6 +56,7 @@ script:
   - docker exec jcsda_container ls
   - docker exec jcsda_container ls /jcsda/src_repo
   - docker exec jcsda_container bash -c 'cp -r /jcsda/.openmpi/ ~/'
+  - docker exec jcsda_container bash -c 'pip install xarray'
```
Comment (mmiesch) [inline .travis.yml:59]: Needed for now but we need to remember to remove this when we move to the new containers

File: buildspec_clang.yml [hunk 2]
```diff
@@ -14,6 +14,10 @@ phases:
       - echo $CODEBUILD_WEBHOOK_ACTOR_ACCOUNT_ID
       - echo $CODEBUILD_WEBHOOK_EVENT
       - echo $CODEBUILD_RESOLVED_SOURCE_VERSION
+      - pip install xarray
+      - echo $PATH
+      - echo $LD_LIBRARY_PATH
+      - echo $PYTHONPATH
```
Comment (mmiesch) [inline buildspec_clang.yml:20]: Do you want to leave these echos in or clean up?  I'm ok either way - leaving them in might be useful in future debugging.

File: buildspec_gnu.yml [hunk 3]
```diff
@@ -96,6 +106,8 @@ phases:
       - if [ "$BUILD_STATUS" = "1" ] && [ "$CODEBUILD_BUILD_SUCCEEDING" = "0" ];
```
Comment (rickgrubin) [inline buildspec_gnu.yml:106]: CodeBuild defines:

`CODEBUILD_BUILD_SUCCEEDING: Whether the current build is succeeding. Set to 0 if the build is failing, or 1 if the build is succeeding.`

Suboptimal that it's opposite of what POSIX defines.

**Pull request merged**